joomla/joomla-cms
# [4.0] Airbnb ECMAScript 2017 (ES6) Coding Standards for the files in directory build/build-modules-js

This PR is part of a javascript modernization plan. See #19643 for more details.

### Summary of Changes
I edited the 6 js-files in directory /build/build-modules-js/ so that this files fullfil the coding standards.

I added three dependencies to the devDependencies of package.json, because we use it and coding standards require this. If I am right, we only use it for build process:
https://github.com/postcss/postcss
https://github.com/petkaantonov/bluebird
https://github.com/chalk/chalk

I added one dependency to the dependencies of package.json, because we use it and coding standards require this. If I am right we use it in atum template.
https://github.com/claviska/jquery-minicolors

I changed the file .eslintignore analogous to this PR: https://github.com/joomla/joomla-cms/pull/19650/files#diff-1dc6ee56b778cd91e0327b52aaeaa1b9R16



### Testing Instructions

1. Apply this patch
2. Run 
npm install
3. Make sure that the coding standards throw no error:
Run
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/*

If you want to check only one file run:
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/compilecejs.js
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/compilecescss.js
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/compilejs.js
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/compilescss.js
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/installation.js
node ./node_modules/eslint/bin/eslint.js build/build-modules-js/update.js

4. Make sure, that the script do still that what they should do

Run the commands
node build –compilecejs
node build –compilececss
node build –compilejs
node build –compilescss
node build –installation
node build –update


--update (Updates the vendor scripts)
--compilejs, --compilejs path (Compiles ES6 to ES5 scripts)
--compilecss, --compilecss path (Compiles all the scss files to css)
--compilecejs, --compilecejs path (Compiles/transpiles all the custom elements files)
--compilecescss, --compilecescss path (Compiles/transpiles all the custom elements files
--installer (Creates the language file for installer error page)


### Expected result
Everything works fine and there are no coding standard errors.


### Actual result
Everything works fine, but there are coding standard errors.
